### DISCUSSION TITLE
Support for Geospatial Information System (GIS) scenario creation

### DISCUSSION THREAD

**Discussion by lkh5kor** [InterDigitalInc/AdvantEDGE#171] [Q&A] [answered]
I am following https://github.com/InterDigitalInc/AdvantEDGE/wiki/gis to create a GIS scenario and downloaded the .mbtiles for Dortmund region. Does <your-installation-folder> mean the AdvantEDGE github repository folder? If yes, then there was no ./meep folder. I even tried creating a empty folder structure as ./meep/omt and copied the downloaded mbtiles file to the folder. But in the map view of scenario in AdvantEDGE, I don't see any network elements(though I add them) nor map tiles (refer to the attached screenshot). Can someone help me on GIS setting up for scenario creation?
![GIS_Scenario](https://user-images.githubusercontent.com/57914213/102880670-fe066500-444b-11eb-8a6a-2eec3f0b06fb.png)

**Comment by nikhildoifode** [comment, marked as answer]
When you deploy the AdvantEDGE, `.meep` folder gets created in your home folder. So I would check the home folder first to see if the `.meep` folder is created there or not. https://github.com/InterDigitalInc/AdvantEDGE/wiki/gis#map-provisioning this link mentions adding the `.mbtiles` file into <your-installation-folder>/.meep/omt. In this case it would be your `home` folder on the Ubuntu system. As I can see that you are able to load the frontend, that means you have deployed the AdvantEDGE successfully. The next steps for you would be to copy the `.mbtiles` file of the Dortmund region to `~/./meep/omt` folder and then delete and deploy the core and dep again.
1. Copy `_.mbtiles` to `~/./meep/omt` folder
2. `meepctl delete core`
3. `meepctl delete dep`
4. `meepctl deploy dep`
5.  `meepctl deploy core`

If your downloaded file is correct you should be able to see the map on the frontend.
FYI, once you have followed the above steps, you should reload the webpage in the browser by ignoring the cache (force reload).

**Reply by lkh5kor** [reply]
Thank you nikhildoifode for the clarification.

After following the above steps, meep-virt-engine goes to error and then CrashLoopBackOff state. I am using AdvantEDGE 1.6 release version. I did try deleting the .mbtiles file and deployed advantEDGE but the meep-virt-engine was still showing error.

Please find the kubectl get pods logs in the attached screenshot. Kindly let me know how I can get the GIS scenario to work.
![kubectl get pods log](https://user-images.githubusercontent.com/57914213/102997013-6759a700-4524-11eb-8f60-a4bc59119f1f.png)

**Reply by roymx** [reply]
Hello @lkh5kor,

Thanks for pointing this issue out, it was caused by a change in an external dependency over the holiday period.
We released a hotfix in v1.6.1 yesterday that addresses this issue.

**Comment by nikhildoifode** [comment]
It would be helpful if you could provide the logs for the meep-virt-engine pods. It's hard inspect the problem without the logs. You could use the `kubectl logs` command for this.
Thank you.

**Reply by lkh5kor** [reply]
Please find the kubectl logs for meep-virt-engine pod below
![kubectl logs mee-virt-engine](https://user-images.githubusercontent.com/57914213/103026473-4e68ea00-4554-11eb-8629-9c7e9c44c05e.png)

**Reply by nikhildoifode** [reply]
I think this problem is related to helm having problem with the proxy connection. This issues doesn't occur every time. My guess is you are using Virtualbox/VMWare or any other kind of Virtual Machine client.

If you are using Virtualbox then in Virtual Machine settings, changing Network Adapter type to Bridged Adaptor from NAT should resolve this problem.
If not then let me know I will try to solve this problem some other way. Also attach the logs too if unsuccessful

**Reply by lkh5kor** [reply]
Thank you for the reply.
I'm not using virtual machine.
I'm using Ubuntu 20.04 operating system as a dual boot alongside windows 10.

**Reply by JeanValjean17** [reply]
Hi @lkh5kor,


I don't know if you solved the problem already, but the solution is to replace the URL in the entrypoint.sh script located in AdvantEdge/go-apps/meep-virt-engine. In the line where this is located:

**helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com/**

it should be: **helm repo add incubator https://charts.helm.sh/incubator** 

After this you need to build again I think and then re-deploy. This should solve the problem. 

Helm apparently does not use that URL anymore, for more info check here: [https://stackoverflow.com/questions/65407317/helm-init-failed-is-not-a-valid-chart-repository-or-cannot-be-reached-failed-to 
](url)

Cheers,
Jean

**Reply by lkh5kor** [reply]
Hi @JeanValjean17 ,

Thank you for sharing the information. After changing the URL in entrypoint.sh, meep-virt-engine pod is running. 

Regards,
Shankar Lokeshwara

**Reply by nikhildoifode** [reply]
Hello @lkh5kor ,
Please let me know if you are able to see the maps in the frontend after helm charts repo fix and then building, dockerizing and deploying again

**Reply by lkh5kor** [reply]
@nikhildoifode and @JeanValjean17 ,

Thank you for the support. Yes, I am able to see the maps now.

**Comment by nikhildoifode** [comment]
Hello,

Recently, the helm charts repository was moved to a new location https://helm.sh/blog/new-location-stable-incubator-charts/
We will push this fix to the repository as soon as possible until then you can add the following fix in the code.
Please change the following line
`helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com/`
to
`helm repo add incubator https://charts.helm.sh/incubator`
in `AdvantEdge/go-apps/meep-virt-engine/entrypoint.sh`

After adding this fix you will have to build, dockerize and deploy again, follow the below steps for that:
`meepctl delete core`
`meepctl build all`
`meepctl dockerize all`
`meepctl deploy core`
